### PR TITLE
Remove old FPIX; make New FPIX default (RFC)

### PR DESCRIPTION
**Pull request by schneiml**
As discussed in a private email thread, this PRR attempts to remove the old FPIX description and replace it with the new one. This will cause a few problems, so this is not ready for merging yet; but it should serve as a proposal for what pre13 could look like. @makortel , @veszpv , @ianna and maybe others might be interested.
- all `/pixfwd/` folders are removed, their content replacing the respective non-pixfwd files.
- A conflict appeared at trackerRecoMaterial.xml; 8a4e78a0bf26f42a55250db1b043ff6f50cebce3 was added after @veszpv forked off the pixfwd version. I attempted a merge (not validated).
- old `PhaseI/pixfwd_*.xml` files removed. This will break `_cfg.py` that use them.
- 2017 and 2017NewFPix config files updated to load the FPIX xmls from their new places. They are now almost identical.
- A conflict appeared at cmsExtendedGeometry2017XML_cfi.py; 2e9f311c changed names after the NewFPix version was forked off. Merged in the new 2017XML_cfi.py; old version in 2017NewFPix.
- 2017NewFPix should be deleted soon.
- Definitly broken cfg files (should be updated if useful, deleted if no longer needed):

```
Geometry/CMSCommonData/python/Phase1_R34F16_cmsSimIdealGeometryXML_cfi.py
Geometry/CMSCommonData/python/cmsExtendedGeometry2017devXML_cfi.py
Geometry/CMSCommonData/python/cmsExtendedGeometry2019XML_cfi.py
Geometry/CMSCommonData/python/cmsExtendedGeometryPhaseIPixelXML_cfi.py
```

Changes validated using MaterialBudget tools to make sure everything is now identical to the old NewFPix geometry. This does **not** check whether the merges are fine.

Further thoughts: This PR will cause a mess for merging the (hopefully coming soon) FPIX material budget improvements. Maybe we want to keep the `/pixfwd/` copy around for a while to validate these changes?

### CODE REVIEW

**Comment by makortel** [comment]
Pardon my ignorance, but does this change affect also the phase2 geometries that use phase1 pixel?

If yes, is that intentional also from phase2 point of view?

If no, should it be done?

@boudoul @ebrondol

**Comment by schneiml** [comment]
@makortel as far as I have discussed with Gaelle, Phase2 does not load anything from Phase1 any longer (since the big Phase2 PR for pre12). Before that, the Phase1 BPIX was directly used by Phase2. For the FPIX, there might still be *copies* of the old FPIX flying around, but those were always copies, so not affected by changes here.

**Comment by boudoul** [comment]
I shall check that first (there was too many changes these days to be completely sure of @schneiml 's statement )- and fix this in case in a subsequent PR

**Comment by makortel** [comment]
@schneiml @boudoul Ok, thanks for the clarifications. So the intention is that the phase2 geometries are (or should become if not yet) fully decoupled from phase1 geometry?

**Comment by slava77** [comment]
@cmsbuild please test
... it wasn't clear from the PR description if this can run at

**Comment by schneiml** [comment]
@makortel yes, I think this is what we want to achieve.

@slava77 Ok, the test results are about as expected. Nothing really broke, just the temporary HCAL dev WF which i did not update. I could do that once we plan to merge this, but lets wait what others propose ( #15994 is related).

**Comment by schneiml** [comment]
I updated the configuration of the failing relval to use the new FPIX. But the problem remains:

- `cmsExtendedGeometryPhaseIPixelXML_cfi.py` is broken and I'd like to remove it, if only for the misleading name. It is used in a few `test/` packages, that might need an update then.
- `cmsExtendedGeometry2019XML_cfi.py` could be fixed and might be interesting, but it is outdated currently anyways. So for a real *real* 2019 scenario, this needs to be re-done based on 2017 anyways.
- For `Phase1_R34F16_cmsSimIdealGeometryXML_cfi.py`, I don't know whether it is still relevant. We keep the specific files around, and it would be possible to move the old Phase1 FPIX there as well. But in terms of general cleanup, removing it is easier...
- There are even more files referring to the old FPIX:
```
SLHCUpgradeSimulations/Geometry/python/Phase1_cmsSimIdealGeometryXML_cfi.py
Geometry/HGCalCommonData/python/testHGCalXML_cfi.py
Geometry/HcalCommonData/python/testPhase1GeometryXML_cfi.py
Geometry/HcalCommonData/python/testPhase2GeometryXML_cfi.py
```
They seem not that important to me, but we'd better ask.

So, I see two options:
1. Give up this PR and leave the Tracker Geometry package in its usual cluttered state. Nothing breaks, but the confusion of multiple similar-but-different geometries that use the same names already shows up. The fact that people might continue using old geometries without knowing does not make things better.
2. Clean up, and remove all the broken things or have them fixed (or have them broken and leave the fixing to the next one who needs it).

Now what to do? 2. is obviously better, but 1. fits the pre13 timeframe better...

(also @ianna @kpedro88 @makortel  @veszpv @boudoul )

**Comment by makortel** [comment]
Regarding the failure in 10624, simplest would be to wait for #15994 since the workflow will be removed there.

**Comment by kpedro88** [comment]
@bsunanda should comment on whether he still needs the test geometries you listed.

**Comment by ianna** [comment]
@schneiml - please, make sure you include overlap fixes and other updates https://github.com/cms-sw/cmssw/pull/16117

**Comment by schneiml** [comment]
@ianna If yours can be merged first I'll adapt this one. This one should also conflict with the FPIX material changes (hopefully coming soon) and resolving the questions probably also takes a few days.

**Comment by ianna** [comment]
@schneiml - sounds good to me, thanks.

@davidlange6 - please, merge #16117 at your earliest convenience, thanks.

**Comment by bsunanda** [comment]
I still need the test configuration in HGCal and HcalCommoinData. It should be modified for the new pixel file.

**Comment by bsunanda** [comment]
I can fix the 3 following test configuration once this PR is integrated:
Geometry/HGCalCommonData/python/testHGCalXML_cfi.py
Geometry/HcalCommonData/python/testPhase1GeometryXML_cfi.py
Geometry/HcalCommonData/python/testPhase2GeometryXML_cfi.py

**Comment by civanch** [comment]
@schneiml , there are conflicts in the branch

**Comment by schneiml** [comment]
Ok, I know handled the problems in a way that seemed reasonable to me (note that I have no idea at all what these geometries were good for):

- `cmsExtendedGeometryPhaseIPixelXML_cfi.py` is deleted, including related files, the single use in DPGAnalysis tools is replaced by real 2017, as in the other tools.
- `cmsExtendedGeometry2019XML_cfi.py` is updated to use the 2017 pixel; but there are probably more things outdated in it.
- `Phase1_R34F16_cmsSimIdealGeometryXML_cfi.py` now has a copy of the pre11 pixel xmls. So it should remain vaguely the same as it was. If there is a concensus that this is not needed anymore, I'd happily delete it as well.
- `SLHCUpgradeSimulations/Geometry/python/Phase1_cmsSimIdealGeometryXML_cfi.py` remains broken (no usages).
- The Hcal/HGcal files remain broken but will be fixed by @bsunanda .

None of the changes are tested, since I have no idea how to use the code that is touched. But the changes seem fairly obvious.

From my side, this looks like something that can be merged; care has to be taken with #16117 .

**Comment by schneiml** [comment]
@civanch I'll check.

**Comment by schneiml** [comment]
rebased to CMSSW_8_1_X at 22c517a23b40824991797bc811509155d3c9584f .

@ianna , can you please check that `Geometry/TrackerCommonData/data/PhaseI/pixfwd.xml` is what you expect? It should be your /pixfwd/pixfwd.xml version, while your PhaseI/pixfwd.xml is gone.

**Comment by civanch** [comment]
please test

File: Geometry/TrackerCommonData/data/PhaseI/R34F16/pixfwd.xml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ <Constant name="Disk2Z"         value="[Disk1Z]+75.0*mm"/> <!-- +72.5*mm--> <!-- +105.0*mm -->
+ <Constant name="Disk3Z"         value="[Disk2Z]+95.0*mm"/> <!-- +120.0*mm-->
+
+ <Constant name="ZCylinder"      value="[AnchorZ]"/>
+
+</ConstantsSection>
+
+<SolidSection label="pixfwd.xml">
+ <Polycone name="PixelForward" startPhi="0*deg" deltaPhi="360*deg" >
+ <ZSection z="[RootStartZ]"  rMin="[cms:TrackBeamR1]" rMax="[RootRadius]"/>
```
Comment (ianna) [inline Geometry/TrackerCommonData/data/PhaseI/R34F16/pixfwd.xml]: @schneiml - I have not checked this scenario, however, I do not see appropriate changes in this file.

**Comment by davidlange6** [comment]
hi all- at this point the PR title implies that we are removing stuff - but I see we manage to add lines of xml in the end. Can you confirm that we are not just duplicating stuff?

**Comment by schneiml** [comment]
Due to 4b67ed3, this PR might actually have added XML. After discussion with @boudoul , I know also deleted the R34F16 geometry which should give a nice negative diff.

This leaves some usages in `SLHCUpgradeSimulations/Geometry` dangling, but the affected configs are in turn unused.

**Comment by schneiml** [comment]
#16149 is a strict superset of this PR with changes that should also go into pre13. Closing this one, the other one is to be used.

**Pull request closed without merging**